processing/src/main/java/io/druid/segment/ColumnValueSelector.java

```diff
@@ -21,7 +21,16 @@
 
 /**
  * Base type for interfaces that manage column value selection, e.g. DimensionSelector, LongColumnSelector
+ *
+ * This interface has methods to get the value in all primitive types, that have corresponding basic aggregators in
+ * Druid: Sum, Min, Max, etc: {@link #getFloat()}, {@link #getDouble()} and {@link #getLong()} to support "polymorphic"
+ * rollup aggregation during index merging.
  */
 public interface ColumnValueSelector
 {
+  float getFloat();
```

Comment:
What do you think about adding a new interface maybe ```PrimitiveColumnSelector``` which extends ```ColumnValueSelector``` and is extended by ```Long/Float/DoubleColumnSelector``` and maybe ```ObjectColumnSelector```? If so, ```DimensionSelector``` can extend ```ColumnValueSelector``` directly and don't have to implement unnecessary methods. I'm not sure what the best name is anyway.